### PR TITLE
WIP: interface for container logs

### PR DESCRIPTION
**Pull request by pranavs18**

### CODE REVIEW

**Comment by pranavs18** [comment]
Container Logs handler added. Need to look into some dependencies.

File: code/implementation/docker/api/src/main/java/io/cattle/platform/docker/api/ExecActionHandler.java [hunk 1]
```diff
@@ -69,7 +69,7 @@ public Object perform(String name, Object obj, ApiRequest request) {
         url.append("://").append(apiAccess.getHostname()).append(":").append(CONSOLE_AGENT_PORT.get());
         url.append(CONSOLE_AGENT_PATH.get());
 
-        return new ContainerExecOutput(url.toString(), apiAccess.getAuthenticationToken());
+        return new HostOutput(url.toString(), apiAccess.getAuthenticationToken());
```
Comment (ibuildthecloud) [inline code/implementation/docker/api/src/main/java/io/cattle/platform/docker/api/ExecActionHandler.java]: rename to HostAccess

File: code/implementation/docker/api/src/main/java/io/cattle/platform/docker/api/model/ContainerLogs.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+import io.github.ibuildthecloud.gdapi.annotation.Field;
+
+import java.util.List;
+
+public interface ContainerLogs {
+
+    @Field(defaultValue="true")
+    boolean getFollow();
+
+    @Field(defaultValue="true")
```
Comment (ibuildthecloud) [inline code/implementation/docker/api/src/main/java/io/cattle/platform/docker/api/model/ContainerLogs.java]: Don't need stdout/stderr

File: code/implementation/docker/api/src/main/java/io/cattle/platform/docker/api/model/ContainerLogs.java [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    boolean getStdout();
+
+    @Field(defaultValue="true")
+    boolean getStdErr();
+    
+    @Field(defaultValue="true")
+    boolean getTimeStamp();
```
Comment (ibuildthecloud) [inline code/implementation/docker/api/src/main/java/io/cattle/platform/docker/api/model/ContainerLogs.java]: don't need timestamp

File: code/implementation/docker/api/src/main/java/io/cattle/platform/docker/api/model/ContainerLogs.java [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+    
+    @Field(defaultValue="100")
+    int getLines();
+
+    @Field(required = true)
+    List<String> getCommand();
```
Comment (ibuildthecloud) [inline code/implementation/docker/api/src/main/java/io/cattle/platform/docker/api/model/ContainerLogs.java]: Don't need command

Comment (pranavs18) [inline code/implementation/docker/api/src/main/java/io/cattle/platform/docker/api/model/ContainerLogs.java]: Yeah, i just realized, will remove this one. Not required here.

File: code/implementation/docker/api/src/main/java/io/cattle/platform/docker/api/ExecActionHandler.java (hunk 1 above)
Comment (pranavs18) [inline code/implementation/docker/api/src/main/java/io/cattle/platform/docker/api/ExecActionHandler.java]: Done.

File: code/implementation/docker/api/src/main/java/io/cattle/platform/docker/api/model/ContainerLogs.java (hunk 2 above)
Comment (pranavs18) [inline code/implementation/docker/api/src/main/java/io/cattle/platform/docker/api/model/ContainerLogs.java]: Removed both of them.

File: code/implementation/docker/api/src/main/java/io/cattle/platform/docker/api/model/ContainerLogs.java (hunk 3 above)
Comment (pranavs18) [inline code/implementation/docker/api/src/main/java/io/cattle/platform/docker/api/model/ContainerLogs.java]: Removed timestamp.

Comment (pranavs18) [inline code/implementation/docker/api/src/main/java/io/cattle/platform/docker/api/model/ContainerLogs.java]: PR updated.

**Pull request closed without merging**